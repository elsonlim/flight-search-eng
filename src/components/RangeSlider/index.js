import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Slider from 'rc-slider';

import 'rc-slider/assets/index.css';

const { createSliderWithTooltip } = Slider;
const Range = createSliderWithTooltip(Slider.Range);

const style = {
	minMax: {
		backgroundColor: '#1678C2',
		color: 'white',
		border: '1px solid #e9e9e9',
		borderRadius: '5px',
		padding: '0 2px',
		fontSize: '0.8em',
		fontWeight: 'normal',
		display: 'flex',
		alignItems: 'center',
		whiteSpace: 'nowrap'
	},
	filterName: { fontWeight: 'bold', marginRight: '10px' },
	rangeInfo: {
		display: 'flex',
		justifyContent: 'space-between',
		flexWrap: 'wrap'
	},
	rangeContainer: { textAlign: 'left', marginBottom: '10px' },
	rangeSlider: {
		container: { padding: '5px' },
		handleStyle: { borderColor: '#1678C2' },
		trackStyle: { backgroundColor: '#1678C2' },
		railStyle: { backgroundColor: 'grey' }
	}
};

/**
 * A RangeSlider object
 */
class RangeSlider extends Component {
	render() {
		const {
			minMax, filterName, rangeInfo, rangeContainer, rangeSlider
		} = style;

		/**
		 * This method generates a slider with infomations.
		 * @returns {ReactElement} a single slider.
		 */
		return (
			<div style={rangeContainer} >
				<div style={rangeInfo}>
					<span style={filterName}>{this.props.name}</span>
					<span style={minMax}>
						{this.props.value.min} - {this.props.value.max}
					</span>
				</div>
				<div style={rangeSlider.container}>
					<Range
						min={this.props.range.min}
						max={this.props.range.max}
						defaultValue={Object.values(this.props.value)}
						handleStyle={[rangeSlider.handleStyle]}
						trackStyle={[rangeSlider.trackStyle]}
						railStyle={rangeSlider.railStyle}
						onAfterChange={newValue =>
							this.props.updateValue(this.props.name, { min: newValue[0], max: newValue[1] })}
					/>
				</div>
			</div>
		);
	}
}

RangeSlider.propTypes = {
	name: PropTypes.string.isRequired,
	value: PropTypes.object.isRequired,
	updateValue: PropTypes.func.isRequired,
};

export default RangeSlider;
