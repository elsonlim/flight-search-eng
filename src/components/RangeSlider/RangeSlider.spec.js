import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import RangeSlider from './index';

describe('RangeSlider', () => {
	let props;

	beforeEach(() => {
		props = {
			name: 'name',
			value: {},
            updateValue: jest.fn(),
            range: {min: 0, max: 3000},
		};
	});

	describe('snapshot test', () => {
		test('renders correctly', () => {
			const tree = renderer.create(<RangeSlider {...props} />).toJSON();
			expect(tree).toMatchSnapshot();
		});
	});
});
