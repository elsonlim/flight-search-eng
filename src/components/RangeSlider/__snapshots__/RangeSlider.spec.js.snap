// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider snapshot test renders correctly 1`] = `
<div
  style={
    Object {
      "marginBottom": "10px",
      "textAlign": "left",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
      }
    }
  >
    <span
      style={
        Object {
          "fontWeight": "bold",
          "marginRight": "10px",
        }
      }
    >
      name
    </span>
    <span
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1678C2",
          "border": "1px solid #e9e9e9",
          "borderRadius": "5px",
          "color": "white",
          "display": "flex",
          "fontSize": "0.8em",
          "fontWeight": "normal",
          "padding": "0 2px",
          "whiteSpace": "nowrap",
        }
      }
    >
       - 
    </span>
  </div>
  <div
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="rc-slider-rail"
        style={
          Object {
            "backgroundColor": "grey",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        className="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;
